src/core/expression/qgsexpressionfunction.cpp

```diff
@@ -804,12 +808,9 @@ static QVariant fcnAggregateGeneric( QgsAggregateCalculator::Aggregate aggregate
                      subExpression,
                      parameters.filter,
                      orderBy );
-  if ( context && context->hasCachedValue( cacheKey ) )
+  if ( context && context->hasCachedValue( cacheKey ) && context->indexOfScope( "Symbol scope" ) == -1 )
```

Comment:
Disabling caching based on the presence of a scope is a workaround for a different issue that *must* be solved on another level.

Reply:
aggregate generic has been restored, an alternative function was made and the context check was moved to the staging functions